### PR TITLE
Adding docs for how to configure Okta for the OIDC provider

### PR DESCRIPTION
**Pull request by mikesiegel**
## Description

This gives step by step instructions on how to configure an Okta provider for use with OIDC and the oauth2 proxy.

## Motivation and Context

By request of the maintainers.

## How Has This Been Tested?

Markdown rendered correctly on github

## Checklist:

- [X ] My change requires a change to the documentation or CHANGELOG.
- [ X] I have updated the documentation/CHANGELOG accordingly.
- [ X] I have created a feature (non-master) branch for my PR.

### CODE REVIEW

File: docs/2_auth.md [hunk 1]
```diff
@@ -144,6 +144,68 @@ OpenID Connect is a spec for OAUTH 2.0 + identity that is implemented by many ma
...
+    "https://example.corp.com"
+]
+email_domains = [
+    "corp.com"
+]
+client_id = "XXXXX"
+client_secret = "YYYYY"
+pass_access_token = true
+cookie_name = "_oauth2_proxy"
+cookie_secret = "ZZZZZ"
```
Comment (ReillyProcentive) [inline docs/2_auth.md]: Can you change this to something that makes it very explicit that the user needs to generate their own cookie secret?

File: docs/2_auth.md [hunk 2]
```diff
@@ -144,6 +144,68 @@ OpenID Connect is a spec for OAUTH 2.0 + identity that is implemented by many ma
...
     -email-domain example.com
 
+The OpenID Connect Provider (OIDC) can also be used to connect to other Identity Providers such as Okta. To configure the OIDC provider for Okta, perform
+the following steps:
+
+#### Configuring the OIDC Provider with Okta
+
+1. Log in to Okta using an administrative account. It is suggested you try this in preview first, `example.oktapreview.com`
+2. Navigate to **Security** then select **API**
+* Click **Add Authorization Server**, if this option is not available you may require an additional license for a custom authorization server.
```
[the remark this answers is not in the document]
Comment (mikesiegel) [reply docs/2_auth.md]: I'll clarify this, it's only necessary if you want to pass a scope for authorization claims on to the protected app: https://developer.okta.com/docs/guides/customize-authz-server/overview/ for authentication, its not necessary.

[the remark this answers is not in the document]
Comment (mikesiegel) [reply docs/2_auth.md]: Sorry clarification. If you just have a single app you're protecting, the default auth server should work, if you want to configure multiple applications with different OAuth2 scopes/claims passed on to each, then it makes this necessary.

File: docs/2_auth.md [hunk 3]
```diff
@@ -144,6 +144,68 @@ OpenID Connect is a spec for OAUTH 2.0 + identity that is implemented by many ma
...
+* Pick an **Application Name** such as `Example App`.
+* Set the **Login redirect URI** to `https://example.corp.com`.
+* Under **General** set the **Allowed grant types** to `Authorization Code` and `Refresh Token`.
+* Leave the rest as default, taking note of the `Client ID` and `Client Secret`.
+* Under **Assignments** select the users or groups you wish to access your application.
+4. Create a configuration file like the following:
+
+```
+provider = "oidc"
+https_address = "localhost:44301"
```
[the remark this answers is not in the document]
Comment (mikesiegel) [reply docs/2_auth.md]: Removed those lines.

File: docs/2_auth.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (mikesiegel) [reply docs/2_auth.md]: Spelled out that it's optional if you are protecting multiple applications and wish to pass different claims/scopes.

File: docs/2_auth.md [hunk 4]
```diff
@@ -144,6 +144,68 @@ OpenID Connect is a spec for OAUTH 2.0 + identity that is implemented by many ma
...
+tls_key_file = "/etc/ssl/server.key"
+tls_cert_file = "/etc/ssl/server.crt"
+redirect_url = "https://example.corp.com"
+oidc_issuer_url = "https://corp.okta.com/oauth2/abCd1234"
```
[the remark this answers is not in the document]
Comment (mikesiegel) [reply docs/2_auth.md]: Removed references.

File: docs/2_auth.md (hunk 1 above)
Comment (mikesiegel) [reply docs/2_auth.md]: Clarified secret needs to be generated.

**Comment by ReillyProcentive** [comment]
LGTM

File: docs/2_auth.md [hunk 5]
```diff
@@ -144,6 +144,66 @@ OpenID Connect is a spec for OAUTH 2.0 + identity that is implemented by many ma
...
+upstreams = [
+    "https://example.corp.com"
+]
+email_domains = [
+    "corp.com"
+]
+client_id = "XXXXX"
+client_secret = "YYYYY"
+pass_access_token = true
+cookie_name = "_oauth2_proxy"
```
Comment (steakunderscore) [inline docs/2_auth.md]: I would be tempted to leave values out of your example config that are the default, like `cookie_name`, `cookie_secure` etc.

Comment (mikesiegel) [reply docs/2_auth.md]: Removed defaults, and removed some extra tweaking stuff not directly related to avoid confusing people.

**Pull request merged**